src/main/java/org/mockito/internal/creation/MockSettingsImpl.java

```diff
@@ -95,8 +98,9 @@ public MockSettings defaultAnswer(Answer defaultAnswer) {
         return this;
     }
 
-    public MockSettings useConstructor() {
+    public MockSettings useConstructor(Object... constructorArgs) {
```

Comment:
+1

@mureinik, can you add a test case for following scenarios:

1. There is a public constructor that takes single parameter and the user creates mock with useConstructor(null).
2. There are multiple overloaded public constructors that takes single parameter of different types, the user creates mock with useConstructor(null)